### PR TITLE
[WIP] Bugfix: add reset SimpleFIN credential button to escape SimpleFIN broken/unable to sync state

### PR DESCRIPTION
**Pull request by mattpetters**
---
category: Bugfix
authors: [mattpetters]
---

Add reset SimpleFIN credential button to resolve a state in which SimpleFIN is unable to sync

### CODE REVIEW

**Comment by psybers** [comment]
Thanks for adding this, I know lots of people have requested it!

It feels quite out of place where you put it though.  To me, it would make more sense in the Advanced Settings screen with the other reset/repair buttons.

**Comment by mattpetters** [comment]
> Thanks for adding this, I know lots of people have requested it!
> 
> It feels quite out of place where you put it though. To me, it would make more sense in the Advanced Settings screen with the other reset/repair buttons.

I actually slept on it and had this same thought, let me move it and I'll update the PR.

**Comment by psybers** [comment]
image

Seems like a good fit here.

**Comment by mattpetters** [comment]
yup

> image
> Seems like a good fit here.

yup, I did more or less the same thing. Feel free to take a look now and lmk if anything else needs maneuvering

**Comment by mattpetters** [comment]
whoops, linter fail

**Comment by mattpetters** [comment]
@psybers ready for another look whenever :)

File: packages/desktop-client/src/components/settings/SimpleFINReset.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          onClick={onSimpleFinInit}
+        >
+          Reset SimpleFIN Credential
+        </ButtonWithLoading>
+      }
+    >
+      {isSimpleFinSetupComplete ? (
+        <Text>
+          <strong>Reset SimpleFIN Credential</strong> In some cases (like
+          importing from an existing database to a new instance), you may need
```
Comment (psybers) [inline packages/desktop-client/src/components/settings/SimpleFINReset.tsx:47]: A bit of a nit, but the other settings incorporate the bolded text into the first sentence.  Here it is acting more like a heading.  Can you reword the text?  Something like:

```
<strong>Reset SimpleFIN credential</strong> if you are having trouble accessing SimpleFIN. ...
```

**Comment by psybers** [comment]
I was not able to actually test the code, as I did not want to deal with resetting my token and do not have time to set up a second/test server.

**Comment by mattpetters** [comment]
> I was not able to actually test the code, as I did not want to deal with resetting my token and do not have time to set up a second/test server.

Probably a hint that I should be updating the docs too. ;) 

On SimpleFIN Bridge, you can add multiple "Apps" that all link to your same account:
Screenshot 2024-05-15 at 2 38 03 PM
Screenshot 2024-05-15 at 2 38 18 PM

Creating that Setup Token will generate the setup token you can enter into the "init" part of the setup and it basically is just a new auth credential to allow you to access your existing SimpleFIN accounts. To my knowledge it should not affect your linked accounts, it's just changing your credential to auth with SimpleFIN.

@psybers

**Comment by mattpetters** [comment]
@psybers actually hold off on this one, I'm going to run some tests of my own and maybe try to write automated tests as well. might not be able to be done done for another day or two

**Comment by mattpetters** [comment]
@psybers the more I sit with this, the more I'm thinking the **right** way to implement this is to actually have this button hit the server and have the server clear the rows instead of just trying to overwrite by forcing setup again. What do you think?

**Comment by psybers** [comment]
@mattpetters I'd suggest throwing a "[WIP]" tag on the PR title while you decide the best course.

**Comment by psybers** [comment]
Since you are rethinking it, maybe look at how the GoCardless sync handles the reset.  I believe it has similar functionality: https://discord.com/channels/937901803608096828/940290142579605514/1232681744780759041

**Comment by MatissJanis** [comment]
👋 We're quite selective about what goes in the "settings" page as we don't want it to get cluttered with all sorts of switches/toggles/buttons. Just the most crucial things.

Take a look at how it's done for gocardless. Ideally we should follow the same pattern for SimpleFin too:

Screenshot 2024-05-16 at 21 38 28
Screenshot 2024-05-16 at 21 40 10

**Comment by psybers** [comment]
We could add a 3-dot menu to the right of the "Link accounts" button:

image

**Comment by MatissJanis** [comment]
> We could add a 3-dot menu to the right of the "Link accounts" button:

Has to be earlier than that IMO. If you currently mis-configure the keys - you won't be able to access the "link accounts" modal. Meaning you wouldn't be able to change the keys.

**Comment by psybers** [comment]
Would it make sense to move the GoCardless one, so then both would be on the initial 'create account' dialog?  Can put two menus there, and maybe only show them if credentials are already configured?

**Comment by mattpetters** [comment]
I think my initial impl was something similar to that with the button below the regular button. Initially I just did that for convenience (everything I wanted to call was right there) but maybe it could have just been refactored to a) be less obtrusive similar to the 3 dot menu and b) only show conditionally as you're suggesting.

PS I should have time to work on this a bit more at some point this weekend so if you guys come upon a design I'll get on it then. And of course feel free to take the reins if you want ;)

**Comment by psybers** [comment]
Maybe something like this? @MatissJanis what do you think?

image

**Comment by MatissJanis** [comment]
I think that looks good (just need to vertically align the "..." menu with the button)!

**Comment by psybers** [comment]
@mattpetters Feel free to use this as a starting point:

https://github.com/psybers/actual/tree/reset-sync

**Comment by MatissJanis** [comment]
👋 just checking in: is this ready for review? If so - would you mind removing the "WIP" prefix from the PR? That lets us know this is ready.

**Comment by psybers** [comment]
@MatissJanis Since this PR still implements it via the settings page, I would guess it is a WIP.

**Pull request closed without merging**